### PR TITLE
Restful WS Spring boot example

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Part of #125

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
Depends on #289

**Review by dsyer — CHANGES_REQUESTED** [review]
Some changes suggested inline.

File: examples/restful-ws-spring-boot/src/main/java/io/cloudevents/examples/springboot/JacksonConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+package io.cloudevents.examples.springboot;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class JacksonConfiguration {
+    @Bean
```
[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply examples/restful-ws-spring-boot/src/main/java/io/cloudevents/examples/springboot/JacksonConfiguration.java:26]: Can I remove `JerseyConfiguration` too?

[the remark this answers is not in the document]
Comment (dsyer) [reply examples/restful-ws-spring-boot/src/main/java/io/cloudevents/examples/springboot/JacksonConfiguration.java]: I think you need to register the `CloudEventProvider`?

[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply examples/restful-ws-spring-boot/src/main/java/io/cloudevents/examples/springboot/JacksonConfiguration.java]: The `CloudEventsProvider` should be registered through the restful ws SPI: https://github.com/cloudevents/sdk-java/tree/master/http/restful-ws/src/main/resources/META-INF/services

[the remark this answers is not in the document]
Comment (dsyer) [reply examples/restful-ws-spring-boot/src/main/java/io/cloudevents/examples/springboot/JacksonConfiguration.java]: I see. Then I guess we still need it to register the `MainResource` at least? It's more common in a Spring Boot app to register the instance, not the class, but in this case it doesn't matter.

[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply examples/restful-ws-spring-boot/src/main/java/io/cloudevents/examples/springboot/JacksonConfiguration.java]: TBH I'm not at all expert of conventions in Spring Boot, so I can't tell which solution is best... Should I change the registration with the instances?

[the remark this answers is not in the document]
Comment (dsyer) [reply examples/restful-ws-spring-boot/src/main/java/io/cloudevents/examples/springboot/JacksonConfiguration.java]: I don't think it matters actually. This sample does it with class literals: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-jersey/src/main/java/smoketest/jersey/JerseyConfig.java

**Comment by salaboy** [comment]
@slinkydeveloper @dsyer this is good! and I think that is basically doing what I need as well.

**Comment by dsyer** [comment]
@salaboy good to hear. Your PR was using Spring `@RestController` though, and I think we should have some support for that as well eventually. @olegz is working on a PR that has a `@RestController` BTW.

**Comment by salaboy** [comment]
@dsyer that's cool.. I think that if webflux is supported I am more than ok with it.. I've created the @RestController stuff to create a PR to this repo, because I was aware that not everyone was into Webflux still. I am happy to see that supported in a much cleaner way.. I believe that this is a big enabler for people to use CE.

**Pull request merged**